openhands-sdk/openhands/sdk/conversation/state.py

```diff
@@ -191,26 +216,42 @@ def create(
 
         # ---- Resume path ----
         if base_text:
-            state = cls.model_validate(json.loads(base_text))
+            persisted_state = cls.model_validate(json.loads(base_text))
 
-            # Enforce conversation id match
-            if state.id != id:
+            if persisted_state.id != id:
                 raise ValueError(
                     f"Conversation ID mismatch: provided {id}, "
-                    f"but persisted state has {state.id}"
+                    f"but persisted state has {persisted_state.id}"
                 )
 
-            # Reconcile agent config with deserialized one
-            resolved = agent.resolve_diff_from_deserialized(state.agent)
+            # Limit to the tools that match - they may have been used in conversation history.
+            # All other config (LLM, agent_context, condenser, etc.) can change freely.
+            agent.load(persisted_state.agent)
+
+            state = cls(
+                id=id,
+                agent=agent,
+                workspace=workspace,
+                persistence_dir=persistence_dir,
+                max_iterations=max_iterations,
+                stuck_detection=stuck_detection,
+                # Restore persisted conversation metadata
+                execution_status=persisted_state.execution_status,
+                confirmation_policy=persisted_state.confirmation_policy,
+                security_analyzer=persisted_state.security_analyzer,
+                activated_knowledge_skills=persisted_state.activated_knowledge_skills,
+                blocked_actions=persisted_state.blocked_actions,
+                blocked_messages=persisted_state.blocked_messages,
+                secret_registry=persisted_state.secret_registry,
+            )
```

Comment:
Instead of creating this from scratch, can we just explicitly overwrite the few keys for `persisted_states`, in case we add more fields in the future but forgot to add a new key here